client-linuxapp/src/main.rs

```diff
@@ -446,37 +484,62 @@ async fn main() -> Result<()> {
     }
 
     // Get rv entries
-    let rv_info = get_rv_info(dc.as_ref());
-
-    let client_list = get_client_list(rv_info.unwrap())
-        .await
-        .context("Error getting usable rendezvous client")?;
+    let rv_info = get_rv_info(dc.as_ref())?;
+    let mut rv_info_it = rv_info.iter().cycle();
 
-    // Get owner info
-    let to1d = get_to1d(dc.as_ref(), client_list).await?;
+    let mut rv_entries_processed: usize = 0;
+    loop {
+        let rv_entry = rv_info_it
+            .next()
+            .ok_or_else(|| anyhow!("No RV entries returned from the iterator"))?;
 
-    let to1d_payload: UnverifiedValue<TO1DataPayload> = to1d
-        .get_payload_unverified()
-        .context("Error getting the TO2 payload")?;
-    let to2_addresses = to1d_payload.get_unverified_value().to2_addresses();
-    let to2_addresses = get_to2_urls(to2_addresses);
-    log::info!("Got TO2 addresses: {:?}", to2_addresses);
+        let client_list = get_client_list(rv_entry)
+            .await
+            .context("Error getting usable rendezvous client")?;
+        log::trace!("Client list: {:?}", client_list);
+
+        // Get owner info
+        let to1d = get_to1d(dc.as_ref(), client_list).await;
+        if let Err(e) = to1d {
+            log::trace!("Error getting TO1d {:?} with rv_entry {:?}", e, rv_entry);
+            rv_entries_processed =
+                sleep_if_needed(rv_entries_processed, rv_info.len(), rv_entry.delay);
+            continue;
+        }
+        let to1d_payload: UnverifiedValue<TO1DataPayload> = to1d?
+            .get_payload_unverified()
+            .context("Error getting the TO1 payload")?;
 
-    if to2_addresses.is_empty() {
-        bail!("No valid TO2 addresses received");
-    }
+        // Contact owner and perform ownership transfer
+        let to2_addresses = to1d_payload.get_unverified_value().to2_addresses();
+        let to2_addresses = get_to2_urls(to2_addresses);
+        log::info!("Got TO2 addresses: {:?}", to2_addresses);
 
-    for to2_address in to2_addresses {
-        match perform_to2(devcred_location.borrow(), dc.as_ref(), &to2_address)
-            .await
-            .context("Error performing TO2 ownership protocol")
-        {
-            Ok(_) => break,
-            Err(e) => {
-                log::trace!("{:?} with TO2 address {}", e, to2_address);
-                continue;
+        if to2_addresses.is_empty() {
+            bail!("No valid TO2 addresses received");
```

Comment:
In this case, we need to continue to the next RV_info entry.

Reply:
Ok, so instead of bailing, logging the error + `continue`?